server/master/src/main/java/org/apache/accumulo/master/tableOps/tableImport/ImportTable.java

```diff
@@ -127,7 +137,50 @@ public void checkVersions(Master env) throws AcceptableThriftTableOperationExcep
 
   @Override
   public void undo(long tid, Master env) throws Exception {
-    Utils.unreserveHdfsDirectory(env, new Path(tableInfo.exportDir).toString(), tid);
+    for (ImportedTableInfo.DirectoryMapping dm : tableInfo.directories) {
+      Utils.unreserveHdfsDirectory(env, new Path(dm.exportDir).toString(), tid);
+    }
+
     Utils.unreserveNamespace(env, tableInfo.namespaceId, tid, false);
   }
+
+  static List<ImportedTableInfo.DirectoryMapping> parseExportDir(String exportDir) {
+    if (exportDir == null || exportDir.isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    String[] exportDirs = exportDir.split(",");
+    List<ImportedTableInfo.DirectoryMapping> dirs = new ArrayList<>(exportDirs.length);
+    for (String ed : exportDirs) {
+      ImportedTableInfo.DirectoryMapping dir = new ImportedTableInfo.DirectoryMapping();
+      dir.exportDir = ed;
+      dirs.add(dir);
+    }
+    return dirs;
+  }
+
+  Path findExportFile(Master env) throws AcceptableThriftTableOperationException {
+    Path exportFilePath = null;
+    for (ImportedTableInfo.DirectoryMapping dm : tableInfo.directories) {
+      exportFilePath = new Path(dm.exportDir, Constants.EXPORT_FILE);
+      try {
+        if (env.getFileSystem().exists(exportFilePath)) {
+          break;
```

Comment:
Would it make sense to refactor this code a bit so that it ensure there is onle one export file in all of the dirs?

Reply:
So, instead of choosing the first one it finds, ensure that there is only one and error if none or more than one is found. Yes, this makes sense.